### PR TITLE
Cleanup NumericsHelpers.DangerousMakeTwosComplement

### PR DESCRIPTION
**Pull request by axelheer**
* Remove "weird" code since it's not in use anyway
* Simplify rest of it to just do the required complement

### CODE REVIEW

**Comment by axelheer** [comment]
cc @stephentoub 

see #3117

File: src/System.Runtime.Numerics/src/System/Numerics/NumericsHelpers.cs [hunk 1]
```diff
@@ -111,49 +111,26 @@ public static double GetDoubleFromParts(int sign, int exp, ulong man)
             return du.dbl;
         }
 
+        // Do an in-place twos complement. "Dangerous" because it causes
+        // a mutation and needs to be used with care for immutable types.
```
Comment (jasonwilliams200OK) [inline src/System.Runtime.Numerics/src/System/Numerics/NumericsHelpers.cs]: nit: *two's* complement :)

**Comment by stephentoub** [comment]
Test Innerloop OSX Debug Build and Test please (https://github.com/dotnet/corefx/issues/4856)

File: src/System.Runtime.Numerics/src/System/Numerics/NumericsHelpers.cs [hunk 2]
```diff
@@ -111,49 +111,26 @@ public static double GetDoubleFromParts(int sign, int exp, ulong man)
...
+        public static void DangerousMakeTwosComplement(uint[] d)
+        {
+            Debug.Assert(d != null);
+            Debug.Assert(d.Length > 0);
```
Comment (stephentoub) [inline src/System.Runtime.Numerics/src/System/Numerics/NumericsHelpers.cs]: What guarantees this is true?

**Comment by stephentoub** [comment]
One question, otherwise LGTM.

File: src/System.Runtime.Numerics/src/System/Numerics/NumericsHelpers.cs (hunk 1 above)
Comment (axelheer) [inline src/System.Runtime.Numerics/src/System/Numerics/NumericsHelpers.cs]: correct :)

File: src/System.Runtime.Numerics/src/System/Numerics/NumericsHelpers.cs (hunk 2 above)
Comment (axelheer) [inline src/System.Runtime.Numerics/src/System/Numerics/NumericsHelpers.cs]: Maybe I assume too much. I change that for safety's sake.

**Pull request merged**